File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxPro =0;
        int minPrise = INT_MAX;
        int n = prices.size();
        for(int i=0;i<n;i++){
            minPrise = min( minPrise, prices[i]);
            maxPro = max(maxPro, prices[i]-minPrise );

        }
        return maxPro;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let tillNow = Number.POSITIVE_INFINITY, ans = 0;
    for(let i=0;i<prices.length;i++){
        tillNow = Math.min(tillNow, prices[i]);
        ans = Math.max(ans, prices[i]-tillNow);
    }
    return ans
};